src/lib/helpers/uploader.helper.ts

```diff
@@ -1,36 +1,54 @@
 import imageCompression from 'browser-image-compression';
 import { keccak256 } from 'ethers/lib/utils';
+import imageSize from 'image-size';
 import { AddResult } from 'ipfs-core-types/src/root';
 import { ImportCandidate } from 'ipfs-core-types/src/utils';
 import { create, Options } from 'ipfs-http-client';
+import sharp from 'sharp';
 
 import { LSP3ProfileImage } from '../interfaces';
+import { ImageBuffer } from '../interfaces/lsp3-profile';
 import { ProfileUploadOptions } from '../interfaces/profile-upload-options';
 
 export const sizes = [1800, 1024, 640, 320, 180];
 export async function imageUpload(
-  givenFile: File,
+  givenFile: File | ImageBuffer,
   uploadOptions: ProfileUploadOptions
 ): Promise<LSP3ProfileImage[]> {
-  const isImage = givenFile.type?.substr(0, 6) === 'image/';
+  const type = 'type' in givenFile ? givenFile.type : givenFile.mimeType;
+  const isImage = type?.substr(0, 6) === 'image/';
   if (!isImage) {
-    throw new Error(`File provided is of type "${givenFile.type}".`);
+    throw new Error(`File provided is of type "${type}".`);
   }
 
   return Promise.all(
     sizes.map(async (size) => {
-      const img = await imageCompression(givenFile, {
-        maxWidthOrHeight: size,
-        useWebWorker: true,
-      });
+      let imgToUpload, imgBuffer, width: number, height: number;
 
-      const imgBuffer = new Uint8Array(await img.arrayBuffer());
-      const loadedImg = await imageCompression.drawFileInCanvas(img);
-      const uploadResponse = await ipfsUpload(img, uploadOptions.ipfsClientOptions);
+      if ('buffer' in givenFile) {
+        imgBuffer = await resizeBuffer(givenFile.buffer, size);
+        height = size;
+        width = size;
+
+        imgToUpload = imgBuffer;
+      } else {
+        imgToUpload = await imageCompression(givenFile, {
+          maxWidthOrHeight: size,
```

Comment:
Can this fail? Should be use try/catch here to prevent errors? (Didn't check the whole flow, just thinking about this potential issue)